aws/corehandlers/handlers.go

```diff
@@ -156,7 +166,25 @@ var AfterRetryHandler = request.NamedHandler{Name: "core.AfterRetryHandler", Fn:
 
 	if r.WillRetry() {
 		r.RetryDelay = r.RetryRules(r)
-		r.Config.SleepDelay(r.RetryDelay)
+
+		if sleepFn := r.Config.SleepDelay; sleepFn != nil {
```

Comment:
It seems like if a `SleepDelay` function is specified and someone canceled the request, that `SleepDelay` will always win. Is that what we want?